server/rest/src/main/java/org/infinispan/rest/cachemanager/RestCacheManager.java

```diff
@@ -71,14 +71,16 @@ public RestCacheManager(EmbeddedCacheManager instance, Predicate<? super String>
          throw logger.missingRequiredMediaType(name);
       }
       checkCacheAvailable(name);
-      String cacheKey = name + "-" + keyContentType.getTypeSubtype() + valueContentType.getTypeSubtype();
+      String mapKey = keyContentType.toString();
+      String mapValue = valueContentType.toString();
```

Comment:
`HotRodServer` avoids the `toString()` calls by keeping a map cacheName -> CacheInfo, and in the CacheInfo a map `KeyValuePair<MediaType, MediaType>` -> encoded cache. 

The only gotcha is that `MediaType.toString()` ignores the weight parameter, while `equals()` and `hashCode()` do not.  I'd rather we remove the weight parameter explicitly instead of ignoring it in `toString()`. FWIW `HotRodServer` ignores all parameters when creating the encoded cache (with `getTypeSubType()`), which is probably wrong.

Reply:
ok, let me change it

Reply:
Ok, I gave up of excluding this ```q``` parameter, it is almost never sent, only when interacting with the REST API from the browser as a GET (not XHR)